LPS-114186 Migrate AUI tree-view in asset-taglib to React

mateomustapic opened the pull request:
Hey @wincent ,

This PR is follow up on

julien#284
julien#285
mateomustapic#9
mateomustapic#10
and
https://github.com/wincent/liferay-portal/pull/313

This PR is a work on migration of AUI tree-view to React in asset-taglib module.
The old aui tree-view implementation had completely different logic than React TreeView requires, so I Eudaldo created new Java classes which can now be used in new implementation.

I used the `<react:component>` tag and included React component in the `jsp` where old tree-view was located, and I additionally cleaned up unnecessary jsp and `AUI` code which was used in old implementation.
Additionally, Julien made modifications to react component for passing `categories` and `vocabularies` to `AssetCategoriesNavigationTreeView` component.

After Eudaldo's final changes in https://github.com/wincent/liferay-portal/pull/313 I modified JS to:
- Fix bug with 'ghost' icons in `TreeView` component 
- Fix categories navigate `url`
- Fix error when clicking on a `vocabulary` and redirect only if `category` selected

To test these changes :

Navigate to Categorization / Categories
Create some categories in Vocabularies list for test
Create a widget page (Site Builder/Pages new Blank Page)
Add the portlet categories navigation to a page
Now the tree view should be visible on created page.

If you need anything else to test this, please tell me.

I am sending screenshot of `TreeView` with fixed "missing icons" bug.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

mateomustapic commented:
<img width="1483" alt="Screenshot at Jun 04 15-08-21" src="https://user-images.githubusercontent.com/25637907/83762561-f29e4880-a677-11ea-803e-ed98c07149af.png">

mateomustapic commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-6.liferay.com/job/test-portal-source-format/3523/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-6.liferay.com//userContent/jobs/test-portal-source-format/builds/3523/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/1009dd3a8aa6d40f01eb56c9099cda7cea9063e1">1009dd3a8aa6d40f01eb56c9099cda7cea9063e1</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/mateomustapic/liferay-portal/tree/LPS-114186-0406">LPS-114186-0406</a><br/>Branch GIT ID: <a href="https://github.com/mateomustapic/liferay-portal/commit/3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf">3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-6.liferay.com/job/test-portal-source-format/3523/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/3523/jenkins-report.html">here</a>.</h5></details></html>

mateomustapic commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest(master)/7435/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-3.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7435/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 19 out of 19 jobs passed</h3><h3>:x: ci:test:relevant - 58 out of 60 jobs passed in 2 hours 3 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/1009dd3a8aa6d40f01eb56c9099cda7cea9063e1">1009dd3a8aa6d40f01eb56c9099cda7cea9063e1</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/6781f47bc448a9efe3c4df1eac5c1b8c4fc86c4e">6781f47bc448a9efe3c4df1eac5c1b8c4fc86c4e</a></p><details><summary><strong>ci:test:stable - 19 out of 19 jobs PASSED</strong></summary><details><summary><strong>19 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/391078/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400492/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/395641/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/1</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340305/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/10</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418814/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/11</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274542/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/2</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/391081/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/3</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/322816/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/4</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340306/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/5</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418815/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/6</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/393014/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/7</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/287902/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/410562/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/9</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418812/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/410561/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400273/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/391080/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/410560/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/397295/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 58 out of 60 jobs PASSED</strong></summary><div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest(master)/7435/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/395639/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/16</strong></strike></a></li></ul></div><details><summary><strong>58 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/397294/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/390286/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/391078/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/322815/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/371835/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/397293/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/362968/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149848/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/12</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/393012/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/13</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400272/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/14</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400490/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/15</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/265358/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/17</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/270380/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/398827/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/390284/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/287901/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149847/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/397292/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/362966/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/391079/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400489/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/398826/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/395640/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400491/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/410559/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418813/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/265356/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/270379/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/322814/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400271/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/397291/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/362967/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400492/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/395641/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/1</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340305/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/10</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418814/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/11</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274542/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/2</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/391081/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/3</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/322816/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/4</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340306/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/5</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418815/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/6</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/393014/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/7</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/287902/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/410562/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/9</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/265357/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/395638/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/390285/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149850/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/362970/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/418812/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/393013/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/362969/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/410561/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/400273/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/391080/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149849/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/410560/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/397295/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><h5>For more details click <a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7435/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/395639/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/16</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/395639//consoleText">AXIS_VARIABLE=0,label_exp=!master #395639</a><pre><code>record-test-generated-properties:
 [sync-dir] Synchronizing /opt/dev/projects/github/liferay-binaries-cache-2020/.gradle/caches/modules-2/files-2.1 into /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1
 [sync-dir] 2319 files synchronized in 292ms
[beanshell] Email sent to: yi-chen.tsai@liferay.com

BUILD FAILED
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common.xml:11825: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-portal-acceptance-batch.xml:219: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common.xml:8911: Sourced file: inline evaluation of: `` 								 									import com.liferay.jenkins.results.parser.GitHubDevSyncUtil;  . . . '' : TargetError : at Line: 91 : in file: inline evaluation of: `` 								 									import com.liferay.jenkins.results.parser.GitHubDevSyncUtil;  . . . '' : throw e ; 

Target exception: java.lang.NullPointerException</code></pre></div></li></ol></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html>

julien commented on modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_navigation/js/AssetCategoriesNavigationTreeView.js:
```diff
@@ -0,0 +1,58 @@
+
+		if (selectedNodeId) {
+			const categoryUrl = categories[0].url;
+
```
@mateomustapic could you explain the logic here?
It might seem obvious but I'm wondering why we're only looking at the first element's `url`?

mateomustapic replied:
> This is a small detail (and they are important) and I know it's not always easy to find a good commit message, I also sometimes struggle to get it right, but one thing that has helped me improve that is reading my commit messages before sending the PR and using `git commit --amend` or `git rebase -i` to fix things
> 
> In this case
> 
> You previous messages for this PR use this format:
> 
> ISSUE_NUMBER MESSAGE
> 
> In [3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf](https://github.com/julien/liferay-portal/pull/286/commits/3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf) you're using:
> 
> ISSUE_NUMBER - MESSAGE
> (with a -)
> 
> As I said above, it's not a big deal, but when looking at `git log` it stands out.
> 
> And I don't fully understand the commit message

This commit message is not different on purpose. I usually prepare commit messages in editor so I don't have typos or similar, but this editor added this - , which I didn't notice.

mateomustapic replied:
this was a mistake, this variable was wrong and unnecessary.
I am working on fix ASAP.

julien commented on modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_navigation/js/AssetCategoriesNavigationTreeView.js:
```diff
@@ -0,0 +1,58 @@
+		children: categories.filter((category) => {
+			return category.vocabularyId === vocabulary.vocabularyId;
+		}),
+		icon: 'vocabulary',
```
I see that for the categories, we're getting the `icon` directly from java, (see [here](https://github.com/julien/liferay-portal/pull/286/files#diff-1d33ac4040193644a41695693b08f96fR185)) and I was wondering if it would be possible to the same for categories just for consistency

mateomustapic replied:
Fixed in this commit https://github.com/julien/liferay-portal/pull/286/commits/dc537673e0094e0b6417a5daff189461a0cad628

julien replied:
I actually meant to get this from the server, but I don't know how to do that.
So maybe we can just revert this commit if that's OK with you.

mateomustapic replied:
we can do that, sure

julien commented on modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_navigation/js/AssetCategoriesNavigationTreeView.js:
```diff
@@ -0,0 +1,58 @@
+				categories.forEach((category) => {
+					if (category.id === selectedNodeId) {
+						Liferay.Util.navigate(categoryUrl + selectedNodeId);
+					}
```
I'm wondering if this is correct. 
In my case when I click in the first "category", `categoryUrl` + `selectedNodeId` evaluates to `"http://localhost:8080/web/guest/page01/-/categories/36237?p_r_p_categoryId=3623736237"`

mateomustapic replied:
You re right, this was wrong, I modified this and url looks like the old one now 
`http://localhost:8080/web/guest/test-page/-/categories/36734?p_r_p_categoryId=36734`
I will push these changes soon.

mateomustapic commented:
> This is a small detail (and they are important) and I know it's not always easy to find a good commit message, I also sometimes struggle to get it right, but one thing that has helped me improve that is reading my commit messages before sending the PR and using `git commit --amend` or `git rebase -i` to fix things
> 
> In this case
> 
> You previous messages for this PR use this format:
> 
> ISSUE_NUMBER MESSAGE
> 
> In [3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf](https://github.com/julien/liferay-portal/pull/286/commits/3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf) you're using:
> 
> ISSUE_NUMBER - MESSAGE
> (with a -)
> 
> As I said above, it's not a big deal, but when looking at `git log` it stands out.
> 
> And I don't fully understand the commit message

This commit message is not different on purpose. I usually prepare commit messages in editor so I don't have typos or similar, but this editor added this - , which I didn't notice.

mateomustapic commented:
I fixed url when clicking on category. Now it looks like this 
`http://localhost:8080/web/guest/test-page/-/categories/36734?p_r_p_categoryId=36734
`
It has now the same behaviour and navigates to the same url as the old implementation (see screenshot below).

mateomustapic commented:
<img width="1523" alt="old_treeview" src="https://user-images.githubusercontent.com/25637907/83859280-f71d3c80-a71d-11ea-9dda-6b9a7bd7f1f3.png">

wincent commented:
![Screenshot 2020-06-05 -123709-cwBPW5CN@2x](https://user-images.githubusercontent.com/7074/83867186-52a0f780-a729-11ea-9199-7d576bd6847c.png)

Oh, GitHub...

![love](https://user-images.githubusercontent.com/7074/83867304-7ebc7880-a729-11ea-9736-178c6931b304.gif)

mateomustapic commented:
> ![Screenshot 2020-06-05 -123709-cwBPW5CN@2x](https://user-images.githubusercontent.com/7074/83867186-52a0f780-a729-11ea-9199-7d576bd6847c.png)
> 
> Oh, GitHub...
> 
> ![love](https://user-images.githubusercontent.com/7074/83867304-7ebc7880-a729-11ea-9736-178c6931b304.gif)

![s](https://user-images.githubusercontent.com/25637907/83867461-bc210600-a729-11ea-8405-16674ef41092.jpg)

wincent commented:
> The old url implementation navigates to this url, but remains on the same page.
> Please @ealonso take a look at this, when you have time.

I gather from your comments (eg. [here](https://github.com/julien/liferay-portal/pull/286/commits/72e8be3af0aa06653664797af923e4bfa24f01bf#r435833853) and [here](https://github.com/julien/liferay-portal/pull/286/commits/3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf#r435776063)) from a few minutes ago that you're still going to make some more changes to this, so perhaps send a new version of the PR to the liferay-echo user when it's fully ready and mention Eudaldo on that one with a specific description of the things you'd like him to look at once you consider it final? I don't know whether your force push addresses those comments or not, but just to be clear, seeing as I know Euldaldo is pretty busy, we should make it as easy as possible for him to know what needs to be checked and when.

mateomustapic commented:
> > The old url implementation navigates to this url, but remains on the same page.
> > Please @ealonso take a look at this, when you have time.
> 
> I gather from your comments (eg. [here](https://github.com/julien/liferay-portal/pull/286/commits/72e8be3af0aa06653664797af923e4bfa24f01bf#r435833853) and [here](https://github.com/julien/liferay-portal/pull/286/commits/3ba871f4a00f2c25f13dae3f8c7f681eeb5b39cf#r435776063)) from a few minutes ago that you're still going to make some more changes to this, so perhaps send a new version of the PR to the liferay-echo user when it's fully ready and mention Eudaldo on that one with a specific description of the things you'd like him to look at once you consider it final? I don't know whether your force push addresses those comments or not, but just to be clear, seeing as I know Euldaldo is pretty busy, we should make it as easy as possible for him to know what needs to be checked and when.

This force-pushed changes are final. I will now send this PR to liferay-echo user and write down everything in the description.

wincent commented:
> This force-pushed changes are final. I will now send this PR to liferay-echo user and write down everything in the description.

Thanks! I'll close this to avoid confusion.

wincent commented:
ci:close
